tests/glsl-intrinsic/shader-subgroup/shader-subgroup-shuffle.slang

```diff
@@ -0,0 +1,131 @@
+//TEST:SIMPLE(filecheck=CHECK_GLSL):  -allow-glsl -stage compute -entry computeMain -target glsl
+//TEST:SIMPLE(filecheck=CHECK_SPV):  -allow-glsl -stage compute -entry computeMain -target spirv -emit-spirv-directly
+
+// not testing hlsl due to missing impl
+//DISABLE_TEST:SIMPLE(filecheck=CHECK_HLSL): -allow-glsl -stage compute -entry computeMain -target hlsl -DTARGET_HLSL
+// not testing cuda due to missing impl
+//DISABLE_TEST:SIMPLE(filecheck=CHECK_CUDA): -allow-glsl -stage compute -entry computeMain -target cuda -DTARGET_CUDA 
+// not testing cpp due to missing impl
+//DISABLE_TEST:SIMPLE(filecheck=CHECK_CPP):  -allow-glsl -stage compute -entry computeMain -target cpp -DTARGET_CPP
+
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl -emit-spirv-directly
+#version 430
+
+//TEST_INPUT:ubuffer(data=[0], stride=4):out,name=outputBuffer
+buffer MyBlockName2 
+{
+    uint data[];
+} outputBuffer;
+
+layout(local_size_x = 32) in;
+
+__generic<T : __BuiltinLogicalType>
+bool test1ShuffleX() {
+    return true
+        && subgroupShuffle(T(1), 1) == T(1)
+#if !defined(TARGET_CUDA) && !defined(TARGET_HLSL)
```

Comment:
It seems that you can now remove this.
I tested on my local machine and HLSL and CUDA works fine for "subgroupShuffleXor()".

Although the implementation for CUDA and HLSL is missing, I don't see why we want to explicitly remove from the testing with the ugly #if  when they are not needed.

Reply:
(I forgot to share a comment on this) -- I agree, the if'defs should be using the cleaner syntax desired as in the arithmetic test. 

for the subgroupShuffleXor(), it should not compile for HLSL/CUDA, I don't know how it did on your local machine, but there is no implementation -- at a minimum slang will complain about a warning on returning values (and therefore not compile stdlib)